### PR TITLE
refactor: [Viz Refactor] Visualisation Templates

### PR DESCRIPTION
**Pull request by h-kanazawa**
#### Which issue does this close?

- Closes
  - https://learningpool.atlassian.net/browse/LL-38
  - https://learningpool.atlassian.net/browse/LL-39
  - https://learningpool.atlassian.net/browse/LL-40
  - https://learningpool.atlassian.net/browse/LL-41
  - https://learningpool.atlassian.net/browse/LL-42
  - https://learningpool.atlassian.net/browse/LL-43

#### Describe your changes providing screenshots of UI changes if necessary.

- Remove legacy constants, PLATFORMS, POPULARACTIVITIES, QUESTIONANALYSIS, SESSIONS, LINE, and TABLE
- Remove constants TEXT and COMPONENT
- Remove VISUALISATION_TEMPLATE_TITLES
- Remove templateId and templateStage
- Add visualisation types for templates
- Move visualisation type constants to lib from ui

### CODE REVIEW

**Comment by h-kanazawa** [comment]
wait for merging https://github.com/LearningLocker/learninglocker/pull/1410

File: ui/src/containers/WidgetVisualiseCreator/index.js [hunk 1]
```diff
@@ -84,9 +84,9 @@ class WidgetVisualiseCreator extends Component {
             schema={schema}
             id={this.getVisualisationId()}
             displayCount={3}
-            parseOption={model => createDefaultTitleWithIcon(model, model.get('description'))}
-            parseOptionString={model => model.get('description', createDefaultTitle(model))}
```
Comment (h-kanazawa) [inline ui/src/containers/WidgetVisualiseCreator/index.js]: When model.get('description') is empty string `""`, `parseOptionString(model)` will not be a string but an object. That causes displaying `[object Object]` in widget visualisation `SimpleInput`

File: ui/src/components/VisualiseIcon/index.js [hunk 2]
```diff
@@ -1,121 +1,138 @@
...
-          { this.renderIcon() }
-          <h5>{this.getTitle(type)}</h5>
-        </div>
-      )
-    );
-  }
-}
+export default withStyles(styles)(React.memo(VisualiseIcon));
+
+// [Viz Refactor] TODO: Remove VisualiseIconWithTitle after every types are refactored in TypeEditor
```
Comment (h-kanazawa) [inline ui/src/components/VisualiseIcon/index.js]: I left some comments starting with `[Viz Refactor] TODO:`. These are tasks that should be done by the end of Viz Refactor project

File: ui/src/containers/VisualiseRow/index.js [hunk 3]
```diff
@@ -1,33 +0,0 @@
-import React from 'react';
-import classNames from 'classnames';
-import { connect } from 'react-redux';
-import withStyles from 'isomorphic-style-loader/lib/withStyles';
-import { modelsSchemaIdSelector } from 'ui/redux/modules/models';
-import VisualiseIcon from 'ui/components/VisualiseIcon';
-import styles from './visualiserow.css';
-
-const VisualiseRow = connect((state, ownProps) => {
```
Comment (h-kanazawa) [inline ui/src/containers/VisualiseRow/index.js]: This component was not used

File: ui/src/utils/projections.js [hunk 4]
```diff
@@ -1,114 +0,0 @@
-/* eslint-disable import/prefer-default-export */
-export const sections = {
```
Comment (h-kanazawa) [inline ui/src/utils/projections.js]: This file is not imported from any other files.

File: ui/src/containers/WidgetVisualiseCreator/index.js (hunk 1 above)
Comment (ryansmith94) [reply ui/src/containers/WidgetVisualiseCreator/index.js]: Nice catch 👍

**Comment by h-kanazawa** [comment]
```
// mongo on staging
db.visualisations.find(   { $or: [{ 'axesgroup': '' }, { 'axesxValue': '' }, { 'axesyValue': '' }, { 'axesvalue': '' }, { 'axesxQuery': '' }, { 'axesyQuery': '' }, { 'axesquery': '' }] },   { 'axesgroup': 1, 'axesxValue': 1, 'axesyValue': 1, 'axesvalue': 1, 'axesxQuery': 1, 'axesyQuery': 1, 'axesquery': 1} ).pretty()
{ "_id" : ObjectId("5d39a802595e3276460d21db"), "axesvalue" : "" }
{ "_id" : ObjectId("5d39a805595e3276460d21dc"), "axesvalue" : "" }
```
I found invalid record in staging MongoDB.
`axes...` should be valid JSON string, but a few record has empty string `""`. That caused JSON parse error. If I delete the records, the error disappears. However, I don't know why the empty string was there.

File: ui/src/containers/VisualiseForm/StatementsForm/OptionsEditor/index.js [hunk 5]
```diff
@@ -1,17 +1,16 @@
 import React from 'react';
-import { XVSY, LEADERBOARD, COUNTER, STATEMENTS, PIE, FREQUENCY } from 'ui/utils/constants';
+import { XVSY, COUNTER, STATEMENTS, PIE, FREQUENCY } from 'ui/utils/constants';
```
Comment (h-kanazawa) [inline ui/src/containers/VisualiseForm/StatementsForm/OptionsEditor/index.js]: 'ui/utils/constants' should be 'lib/constants/visualise'

**Pull request merged**